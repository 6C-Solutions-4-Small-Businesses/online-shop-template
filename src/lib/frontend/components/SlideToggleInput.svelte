<script lang="ts">
    import {SlideToggle} from '@skeletonlabs/skeleton'

    export let label: string
    export let value: boolean
    export let onChange: (event: Event) => void
</script>

<div class="w-full mb-3">
    <div class="flex justify-between items-center bg-slate-200 p-3.5 text-black rounded-md">
        <p>{label}</p>
        <SlideToggle
                active="bg-orange-400"
                background="bg-slate-400"
                checked={value}
                name="slide"
                on:change={onChange}
                size="sm"
        />
    </div>
</div>