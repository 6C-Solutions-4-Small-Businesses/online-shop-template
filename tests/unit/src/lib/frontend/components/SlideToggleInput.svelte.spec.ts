import {render, type RenderResult} from '@testing-library/svelte'
import SlideToggleInput from '$lib/frontend/components/SlideToggleInput.svelte'
import {expect, vi} from 'vitest'

let rendering: RenderResult<SlideToggleInput, typeof import('@testing-library/dom/types/queries')>


describe('Slide toggle input Component', () => {
    describe('html structure', () => {

        beforeEach(() => {
            rendering = render(SlideToggleInput, {
                label: 'Re-use same address',
                value: true,
                onChange: vi.fn()
            })
        })

        it('should have match snapshot', () => {
            expect(rendering.container).toMatchSnapshot()
        })
    })
})