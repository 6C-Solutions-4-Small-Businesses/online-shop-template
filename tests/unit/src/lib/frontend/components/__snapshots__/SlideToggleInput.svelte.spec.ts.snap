// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Slide toggle input Component > html structure > should have match snapshot 1`] = `
<div>
  <div
    class="w-full mb-3"
  >
    <div
      class="flex justify-between items-center bg-slate-200 p-3.5 text-black rounded-md"
    >
      <p>
        Re-use same address
      </p>
       
      <div
        aria-checked="true"
        aria-label=""
        class="slide-toggle inline-block rounded-full hover:brightness-[105%] dark:hover:brightness-110 cursor-pointer "
        data-testid="slide-toggle"
        id=""
        role="switch"
        tabindex="0"
      >
        <label
          class="slide-toggle-label unstyled flex items-center"
        >
          <input
            class="slide-toggle-input hidden"
            name="slide"
            type="checkbox"
          />
           
          <div
            class="slide-toggle-track flex transition-all duration-[200ms] cursor-pointer  rounded-full w-12 h-6 bg-orange-400"
          >
            <div
              class="slide-toggle-thumb w-[50%] h-full scale-[0.8] transition-all duration-[200ms] shadow rounded-full bg-white/75 translate-x-full"
            />
          </div>
           
        </label>
      </div>
    </div>
  </div>
</div>
`;
